### PR TITLE
scmutils: init at 20230902

### PR DESCRIPTION
**Pull request by fedeinthemix**
## Description of changes

Add the latest  `scmutils` running on `mit-scheme`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sc/scmutils/package.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  mitschemeX11,
+  writeShellScriptBin,
+}:
+stdenv.mkDerivation rec {
+  pname = "scmutils";
+  version = "20230902";
+
+  src = fetchurl {
+    url = "https://groups.csail.mit.edu/mac/users/gjs/6946/mechanics-system-installation/native-code/${pname}-src-${version}.tar.gz";
+    sha256 = "sha256-9/shOxoKwJ4uDTHmvXqhemgy3W+GUCmoqFm5e1t3W0M=";
```
Comment (sikmir) [inline pkgs/by-name/sc/scmutils/package.nix:14]:
```suggestion
    hash = "sha256-9/shOxoKwJ4uDTHmvXqhemgy3W+GUCmoqFm5e1t3W0M=";
```

File: pkgs/by-name/sc/scmutils/package.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    echo "#!$SHELL" > $out/bin/scmutils
+    echo "$CMD" "\$@" >> $out/bin/scmutils
+    echo "#!$SHELL" > $out/bin/edwin-scmutils
+    echo "$CMD" "--edit" "\$@" >> $out/bin/edwin-scmutils
+    chmod uog+rx "$out/bin/scmutils" "$out/bin/edwin-scmutils"
+    ln -r -s "$out/bin/edwin-scmutils" "$out/bin/mechanics"
+  '';
+
+  meta = {
+    description = "Scheme library for mathematical physics.";
```
Comment (sikmir) [inline pkgs/by-name/sc/scmutils/package.nix:50]:
```suggestion
    description = "Scheme library for mathematical physics";
```

File: pkgs/by-name/sc/scmutils/package.nix (hunk 1 above)
Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix:14]: I'll change as suggested.

File: pkgs/by-name/sc/scmutils/package.nix (hunk 2 above)
Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix:50]: I'll fix this.

File: pkgs/by-name/sc/scmutils/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  configurePhase = ''
+    ln -r -s kernel/ghelper-pro.scm kernel/ghelper.scm
+    ln -r -s solve/nnsolve.scm solve/solve.scm
+    substituteInPlace load.scm \
+      --replace-fail '/usr/local/scmutils/' "$out/lib/mit-scheme/"
+  '';
+
+  buildPhase = ''
+    echo '(load "compile")' | mit-scheme --no-init-file --batch-mode --interactive
+    echo '(load "load") (disk-save "edwin-mechanics.com")' | mit-scheme --no-init-file --batch-mode --interactive
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/scmutils/package.nix:28]:
```suggestion
    runHook preBuild
    echo '(load "compile")' | mit-scheme --no-init-file --batch-mode --interactive
    echo '(load "load") (disk-save "edwin-mechanics.com")' | mit-scheme --no-init-file --batch-mode --interactive
    runHook postBuild
```

File: pkgs/by-name/sc/scmutils/package.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/lib/mit-scheme/" "$out/share/scmutils" "$out/bin"
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/scmutils/package.nix:32]:
```suggestion
    runHook preInstall
    mkdir -p "$out/lib/mit-scheme/" "$out/share/scmutils" "$out/bin"
```

File: pkgs/by-name/sc/scmutils/package.nix [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      cp -a "$SRC" "$TARGET"scmutils/"$SRC"
+    done
+    # Convenience script to load the band
+    declare -r CMD="exec ${mitschemeX11}/bin/mit-scheme --band $out/lib/mit-scheme/edwin-mechanics.com"
+    echo "#!$SHELL" > $out/bin/scmutils
+    echo "$CMD" "\"\$@\"" >> $out/bin/scmutils
+    echo "#!$SHELL" > $out/bin/edwin-scmutils
+    echo "$CMD" "--edit" "\"\$@\"" >> $out/bin/edwin-scmutils
+    chmod uog+rx "$out/bin/scmutils" "$out/bin/edwin-scmutils"
+    ln -r -s "$out/bin/edwin-scmutils" "$out/bin/mechanics"
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/scmutils/package.nix:46]:
```suggestion
    ln -r -s "$out/bin/edwin-scmutils" "$out/bin/mechanics"
    runHook postInstall
```

File: pkgs/by-name/sc/scmutils/package.nix (hunk 3 above)
Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix:28]: I'll update the commit to include these changes.

File: pkgs/by-name/sc/scmutils/package.nix (hunk 5 above)
Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix:46]: I'll update the commit to include this change.

File: pkgs/by-name/sc/scmutils/package.nix (hunk 4 above)
Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix:32]: I'll update the commit to include this change.

File: pkgs/by-name/sc/scmutils/package.nix [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    hash = "sha256-9/shOxoKwJ4uDTHmvXqhemgy3W+GUCmoqFm5e1t3W0M=";
+  };
+
+  buildInputs = [ mitschemeX11 ];
+
+  configurePhase = ''
+    ln -r -s kernel/ghelper-pro.scm kernel/ghelper.scm
+    ln -r -s solve/nnsolve.scm solve/solve.scm
+    substituteInPlace load.scm \
+      --replace-fail '/usr/local/scmutils/' "$out/lib/mit-scheme/"
```
Comment (FliegendeWurst) [inline pkgs/by-name/sc/scmutils/package.nix:22]:
```suggestion
    runHook preConfigure
    ln -r -s kernel/ghelper-pro.scm kernel/ghelper.scm
    ln -r -s solve/nnsolve.scm solve/solve.scm
    substituteInPlace load.scm \
      --replace-fail '/usr/local/scmutils/' "$out/lib/mit-scheme/"
    runHook postConfigure
```

Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix:23]: I've updated the commit.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by fedeinthemix** [comment]
@NixOS/nixos-release-managers

**Comment by fedeinthemix** [comment]
ZHF: #352882

File: pkgs/by-name/sc/scmutils/package.nix [hunk 7]
```diff
@@ -0,0 +1,68 @@
+{
+  stdenv,
+  fetchurl,
+  lib,
+  mitschemeX11,
+}:
+stdenv.mkDerivation rec {
```
Comment (RossComputerGuy) [inline pkgs/by-name/sc/scmutils/package.nix:7]: Please use `finalAttrs:` instead of `rec`.

**Review by RossComputerGuy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sc/scmutils/package.nix (hunk 7 above)
Comment (fedeinthemix) [reply pkgs/by-name/sc/scmutils/package.nix]: OK. I'll push an updated commit.

**Comment by RossComputerGuy** [comment]
Result of `nixpkgs-review pr 340184` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scmutils</li>
  </ul>
</details>

**Review by RossComputerGuy — APPROVED** [review]

**Pull request merged**